index.js

```diff
@@ -9,3 +10,33 @@ import ShareExtension from 'share_extension/android';
 if (Platform.OS === 'android') {
     AppRegistry.registerComponent('MattermostShare', () => ShareExtension);
 }
+
+// Uncomment the snippet below if you want to update the modules
+// defined in packager/modulePaths.js so they are included in the main bundle.
+
+/*
+/!* eslint-disable no-console *!/
+/!* eslint-disable no-undef *!/
+if (__DEV__) {
+    const modules = require.getModules();
+    const moduleIds = Object.keys(modules);
+    const loadedModuleNames = moduleIds.
+        filter((moduleId) => modules[moduleId].isInitialized).
+        map((moduleId) => modules[moduleId].verboseName);
+
+    const waitingModuleNames = moduleIds.
+        filter((moduleId) => !modules[moduleId].isInitialized).
+        map((moduleId) => modules[moduleId].verboseName);
+
+    // make sure that the modules you expect to be waiting are actually waiting
+    console.log(
+        'loaded:',
+        loadedModuleNames,
+        'waiting:',
+        waitingModuleNames
+    );
+
+    // grab this text blob, and put it in a file named packager/moduleNames.js
+    console.log(`module.exports = ${JSON.stringify(loadedModuleNames.sort())};`);
+}
+*/
```

Comment:
@enahum would it be possible to automate this?

One thing that I worry about is the manual process required to update the modulePaths.js file.
What if a new change is made to the cold-start process and forget to update the file.

Reply:
I don't think this can be automated as you need to grab the output, move it into a file then run the `generateModulesPath.js` script in order to get the blacklisted modules, if you can think of a way to automate this, then be my guest :)

Reply:
Agreed, it's not a simple process.  I can see running an emulator on an integration server and notifying if any changes are required for updating the modulePaths.js file.


Apart from that can't think of other options

Reply:
Now someone making changes that affect the cold start should be aware of this or someone can make them aware during the review process.

I think that adding more code to the app in other places will affect, but lets wait and see